### DISCUSSION TITLE
List of Tables and Tables of Figures

### DISCUSSION THREAD

**Discussion by ghost** [plutext/docx4j#486] [Q&A] [answered]
Hi,

I was trying to find in the library a way to insert "List of Tables" and "List of Figures"  in a similar way that I do for table of contents. I was searching around and I couldn't find anything on that. It seems as if I have to do it from scratch. 

For table of contents we have:

```
WordprocessingMLPackage docx = WordprocessingMLPackage.createPackage();
TocGenerator tocGen = new TocGenerator(docx);
Toc.setTocHeadingText("My Title");
tocGen.generateToc( 0, " TOC \\o \"1-3\" \\h \\z \\u ", false);
tocGen.updateToc(false);
```

Is there anything similar for "list of tables" and "table of figures"?

**Comment by ahreurink** [comment]
Hey Symeon, did you get anywhere with this? I'm looking for the same thing.

**Reply by ghost** [reply]
Unfortunately not....I created custom in html and inserted to docx without pagination

**Reply by ahreurink** [reply]
Sad to hear but thanks a lot for your response! Sadly this this requires us to use a workaround as well. In our case:

If your system has Word installed you can use powershell [_Document interface](https://learn.microsoft.com/en-us/dotnet/api/microsoft.office.interop.word._document?view=word-pia) to add the tables

**Comment by Mieges** [comment]
Hi Symeon, 
I have a map with lists as values. These contain first names and surnames, for example. I would like to insert all surnames with a variable, as I would like to have all participants for an event listed in a document. 
I think your problem is very similar, isn't it? And docx4j doesn't offer any support for this, right?

**Reply by ahreurink** [reply]
Hello Mieges, it sounds like you are looking for a regular table. I'm sure docx4j supports this. This thread is about the index of all tables of the document.

**Comment by plutext** [comment, marked as answer]
Table of figures is just a TOC with \c flag.

        <w:instrText xml:space="preserve"> TOC \h \z \c "Figure" </w:instrText>

The caption numbers are SEQ field.  Please have a look at http://webapp.docx4java.org/OnlineDemo/ecma376/WordML/SEQ.html  

Now implemented in docx4j, and to be part of a docx4j 11.5.4 release:

- TOC supports the C switch:  https://github.com/plutext/docx4j/blob/VERSION_11_5_4/docx4j-core/src/main/java/org/docx4j/toc/switches/CSwitch.java

- You can renumber captions (SEQ fields):  https://github.com/plutext/docx4j/blob/VERSION_11_5_4/docx4j-core/src/main/java/org/docx4j/model/fields/seq/FieldUpdaterSEQ.java and see example https://github.com/plutext/docx4j/blob/VERSION_11_5_4/docx4j-samples-docx4j/src/main/java/org/docx4j/samples/FieldUpdaterSEQExample.java (you can do this after you have merged documents)

- ARABIC, roman and Roman numbering is supported:  https://github.com/plutext/docx4j/blob/VERSION_11_5_4/docx4j-core/src/main/java/org/docx4j/model/fields/FormattingSwitchHelper.java#L745
